controllers/helmchart_controller.go

```diff
@@ -527,7 +539,16 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 
 		if file != "" {
 			defer func() {
-				os.Remove(file)
+				if err := os.Remove(file); err != nil {
+					log.Error(err, "failed to delete temporary credentials file")
+					r.Eventf(
+						obj,
+						corev1.EventTypeWarning,
+						meta.FailedReason,
+						"failed to delete temporary credentials file: %s",
+						err,
+					)
```

Comment:
We have `eventLogf()` just for this use case, refer https://github.com/fluxcd/source-controller/blob/841ed7ae665a0447f596a125aee50eb1b063c158/controllers/helmchart_controller.go#L1090 .